nixos/modules/services/web-apps/pixelfed.nix

```diff
@@ -0,0 +1,366 @@
+        REDIS_CLIENT=predis
+        REDIS_SCHEME=tcp
+        REDIS_HOST=127.0.0.1
+        REDIS_PASSWORD=null
+        REDIS_PORT=6379
+
+        # Laravel Configuration
+        SESSION_DRIVER=database
+        CACHE_DRIVER=redis
+        QUEUE_DRIVER=redis
+        BROADCAST_DRIVER=log
+        LOG_CHANNEL=stack
+        HORIZON_PREFIX=horizon-
+
+        # ActivityPub Configuration
+        ACTIVITY_PUB=false
+        AP_REMOTE_FOLLOW=false
+        AP_INBOX=false
+        AP_OUTBOX=false
+        AP_SHAREDINBOX=false
+
+        # Experimental Configuration
+        EXP_EMC=true
+
+        ## Mail Configuration (Post-Installer)
+        MAIL_DRIVER=log
+        MAIL_HOST=smtp.mailtrap.io
+        MAIL_PORT=2525
+        MAIL_USERNAME=null
+        MAIL_PASSWORD=null
+        MAIL_ENCRYPTION=null
+        MAIL_FROM_ADDRESS=pixelfed@example.com
+        MAIL_FROM_NAME=Pixelfed
+
+        ## S3 Configuration (Post-Installer)
+        PF_ENABLE_CLOUD=false
+        FILESYSTEM_DRIVER=local
+        FILESYSTEM_CLOUD=s3
+        #AWS_ACCESS_KEY_ID=
+        #AWS_SECRET_ACCESS_KEY=
+        #AWS_DEFAULT_REGION=
+        #AWS_BUCKET=<BucketName>
+        #AWS_URL=
+        #AWS_ENDPOINT=
+        #AWS_USE_PATH_STYLE_ENDPOINT=false
+        '';
+      };
+
+      # database config taken from zabbix.nix
+      database = {
+        type = mkOption {
+          type = types.enum [ "mysql" "pgsql" ];
+          example = "pgsql";
+          default = "mysql";
+          description = lib.mdDoc "Database engine to use.";
+        };
+
+        host = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = lib.mdDoc "Database host address.";
+        };
+
+        port = mkOption {
+          type = types.int;
+          default =
+            if cfg.database.type == "mysql" then (head config.services.mysql.settings.mysqld.port)
+            else config.services.postgresql.port;
+          defaultText = literalExpression ''
+            if config.services.pixelfed == "mysql" then config.services.mysql.port
+            else config.services.postgresql.port
+          '';
+          description = lib.mdDoc "Database host port.";
+        };
+
+        name = mkOption {
+          type = types.str;
+          default = "pixelfed";
+          description = lib.mdDoc "Database name.";
+        };
+
+        user = mkOption {
+          type = types.str;
+          default = "pixelfed";
+          description = lib.mdDoc "Database user.";
+        };
+
+        password = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = lib.mdDoc ''
+            The database user's password.
+          '';
+        };
+
+      };
+
+      appKey = mkOption {
+        type = types.str;
+        description = lib.mdDoc '' A random
+          32-character string to be used as an encryption key. No default value;
+          use php artisan key:generate in the dataDir to generate. '';
```

Comment:
Can I use any 32 char string? Likely no one has artisan locally before configuring this.

Reply:
Yes, `openssl rand -hex 32` works fine.

Reply:
Will update the doc.